include/oneapi/dpl/pstl/hetero/algorithm_impl_hetero.h

```diff
@@ -49,7 +49,9 @@ __pattern_walk1(_ExecutionPolicy&& __exec, _ForwardIterator __first, _ForwardIte
         oneapi::dpl::__ranges::__get_sycl_range<__par_backend_hetero::access_mode::read_write, _ForwardIterator>();
     auto __buf = __keep(__first, __last);
 
-    oneapi::dpl::__par_backend_hetero::__parallel_for(__exec, unseq_backend::walk_n<_ExecutionPolicy, _Function>{__f},
+    using __backend_tag_type = typename decltype(__tag)::__backend_tag;
```

Comment:
Why don't we use the same approach for host backend dispatching (TBB/OpenMP/ThreadPool)? It would be looked more flexibility that a "macro switching", IMHO.

Reply:
because here we have different policy types. Basing on those we may go with different decisions about the backend tag. Unfortunately for all the host policies (so far) we have the same set of types (std::sequenced_policy, std::parallel_policy and so forth) independently on the backend (TBB/OpenMP/etc.), thus we cannot make a decision just basing on policy type.

Reply:
1) That's why I'm asking about a special tag. Yes, we cannot differ the host parallel backends by policy, but we can by a tag.
2) Curiosity fact about the hetero part: we can differ by policy, but we don't do it and introduce a special tag:
```
__par_backend_hetero::__parallel_find(__backend_tag_type{},
        ::std::forward<_ExecutionPolicy>(__exec) 
```

Reply:
Even if it's not necessary right away if we would like to keep two different backends coexistence it might be worth to add the second-level tag.